trio/_core/_run.py

```diff
@@ -1293,6 +1297,9 @@ class Runner:
     is_guest = attr.ib(default=False)
     guest_tick_scheduled = attr.ib(default=False)
 
+    def __attrs_post_init__(self):
+        self.asyncgens.install_hooks(self)
+
```

Comment:
Can we do this in `setup_runner` instead? I know the current structure is a bit weird and maybe should be refactored, but I still think we should put all the setup code in one place so when we get around to the refactoring we can find it all easily :-)